### ISSUE TITLE
SSM Agent Installation fails if already installed

### ISSUE THREAD

**Issue by pjaudiomv** [awslabs/amazon-eks-ami#1500]
**What happened**:
When custom AMI already has SSM agent the packer build fails

**What you expected to happen**:
skip the package gracefully

**How to reproduce it (as minimally and precisely as possible)**:
Use CIS Amazon Linux 2 Benchmark - STIG as base AMI
Returns Error:
```
2023-10-31T14:04:19-04:00:     amazon-ebs: Examining /var/tmp/yum-root-DVDoG9/amazon-ssm-agent.rpm: amazon-ssm-agent-3.2.1705.0-1.x86_64
2023-10-31T14:04:19-04:00:     amazon-ebs: /var/tmp/yum-root-DVDoG9/amazon-ssm-agent.rpm: does not update installed package.
2023-10-31T14:04:19-04:00:     amazon-ebs: Error: Nothing to do
```

**Anything else we need to know?**:

This worked until [v20230816](https://github.com/awslabs/amazon-eks-ami/blob/v20230816/scripts/install-worker.sh#L484-L489) release.

If I just run ` sudo yum install -y amazon-ssm-agent` in `install-worker.sh` the build runs fine. It doesn't not when specifying an RPM package `sudo yum install -y https://s3.${BINARY_BUCKET_REGION}.${S3_DOMAIN}/amazon-ssm-${BINARY_BUCKET_REGION}/${SSM_AGENT_VERSION}/linux_${ARCH}/amazon-ssm-agent.rpm`

This relates to https://github.com/awslabs/amazon-eks-ami/issues/685

Maybe just adding a check for agent or/and agent version would suffice.

**Environment**:
- AWS Region: us-gov-west-1
- Instance Type(s): m5.large
- EKS Platform version (use `aws eks describe-cluster --name <name> --query cluster.platformVersion`): 
- Kubernetes version (use `aws eks describe-cluster --name <name> --query cluster.version`): 1.27.6
- AMI Version: AL2 Stig AMI
- Kernel (e.g. `uname -a`):
- Release information (run `cat /etc/eks/release` on a node):

```
```

**Comment by cartermckinnon** [comment]
Feel free to open a PR, you can do a check like this to only install when necessary: https://github.com/awslabs/amazon-eks-ami/blob/7e6c213f14f1dd38551a7589c332c034161ad43f/scripts/install-worker.sh#L96-L97

**Comment by pjaudiomv** [comment]
> Feel free to open a PR, you can do a check like this to only install when necessary:
> 
> https://github.com/awslabs/amazon-eks-ami/blob/7e6c213f14f1dd38551a7589c332c034161ad43f/scripts/install-worker.sh#L96-L97

Thanks just did here https://github.com/awslabs/amazon-eks-ami/pull/1501

**Comment by pjaudiomv** [comment]
Fixed with merging of #1501
